### PR TITLE
Introduce toMajor method for currency conversion

### PR DESCRIPTION
**Pull request by daisy1754**
# Description
This PR introduces utility function toMajor to convert minor (cents) to major (dollar)

old code: $amount / 100
new code: toMajor($amount, "USD")

Note we still hard-code "USD" (so this PR shouldn't change any behaviour). In subsequent PRs I'll replace hardcoded "USD" into appropriate currency code.
See also: https://github.com/BoltApp/bolt-magento2/pull/450
Fixes: https://app.asana.com/0/941920570700290/1130539126613441/f

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Helper/Order.php [hunk 1]
```diff
@@ -417,13 +417,14 @@ private function setPaymentMethod($quote)
...
+     * @throws \Exception
      */
     private function adjustTaxMismatch($transaction, $order, $quote)
     {
-        $boltTaxAmount = round($transaction->order->cart->tax_amount->amount / 100, 2);
-        $boltTotalAmount = round($transaction->order->cart->total_amount->amount / 100, 2);
-
-        $orderTaxAmount = round($order->getTaxAmount(), 2);
+        $precision = CurrencyUtils::getPrecisionForCurrencyCode("USD");
+        $boltTaxAmount = round(CurrencyUtils::toMajor($transaction->order->cart->tax_amount->amount, "USD"), $precision);
```
Comment (roopakv) [inline Helper/Order.php]: can we make toMajor round to the precision?

File: Model/Api/ShippingMethods.php [hunk 2]
```diff
@@ -680,7 +680,7 @@ public function getShippingOptions($quote, $addressData)
 
             $diff = CurrencyUtils::toMinorWithoutRounding($cost, "USD") - $roundedCost;
 
-            $taxAmount = $this->cartHelper->getRoundAmount($shippingAddress->getTaxAmount() + $diff / 100);
+            $taxAmount = round(CurrencyUtils::toMinorWithoutRounding($shippingAddress->getTaxAmount(), "USD") + $diff);
```
Comment (daisy1754) [inline Model/Api/ShippingMethods.php]: I slightly modified this but logic should be more or less same.
previously it was
`round(getTaxAmount() + diff / 100) * 100)`
now it is
`round(getTaxAmount() * 100 + diff)`

File: Helper/Order.php (hunk 1 above)
Comment (daisy1754) [reply Helper/Order.php]: done

File: Model/Api/ShippingMethods.php (hunk 2 above)
Comment (roopakv) [reply Model/Api/ShippingMethods.php]: was it 
`round((getTaxAmount() + diff / 100) * 100)` ? or was it 
`round(getTaxAmount() + diff / 100) * 100;`

just confirming so we dont get rounding error

Comment (daisy1754) [reply Model/Api/ShippingMethods.php]: sorry it was `round((getTaxAmount() + diff / 100) * 100)` (the former)

**Pull request merged**